a3fe/run/stage.py

```diff
@@ -229,24 +222,29 @@ def lam_windows(self) -> _List[_LamWindow]:
         return self._sub_sim_runners
 
     @lam_windows.setter
-    def legs(self, value) -> None:
+    def lam_windows(self, value) -> None:
         self._logger.info("Modifying/ creating lambda windows")
         self._sub_sim_runners = value
 
+    @property
+    def get_lambda_values(self) -> _List[float]:
+        """Return list of lambda values from the engine configuration."""
+        return _SomdConfig._from_config_file(_os.path.join(self.input_dir, "somd.cfg")).lambda_values
+
     @property
     def lam_val_weights(self) -> _List[float]:
         """Return the weights for each lambda window. These are calculated
         according to how each windows contributes to the overall free energy
         estimate, as given by TI and the trapezoidal rule."""
         lam_val_weights = []
-        for i, lam_val in enumerate(self.lam_vals):
+        for i, lam_val in enumerate(self.engine_config.lambda_values):
```

Comment:
If lam_vals is a property which reads self.engine_config, no need to change this. This may be clearer though - so up to you.